core/dbt/adapters/base/query_headers.py

```diff
@@ -77,13 +77,14 @@ def __init__(self, config: AdapterRequiredConfig):
         self.generator: QueryStringFunc = lambda name, model: ''
         # if the comment value was None or the empty string, just skip it
         if comment_macro:
+            assert isinstance(comment_macro, str)
```

Comment:
mypy isn't smart enough to know that if `comment_macro` was `None`, this block would be unreachable.